java-agent/instrumentation/servlet/request-3/src/main/java/io/opentelemetry/auto/instrumentation/servlet3/Servlet3Advice.java

```diff
@@ -24,52 +22,54 @@
 public class Servlet3Advice {
 
   @Advice.OnMethodEnter(suppress = Throwable.class)
-  public static AgentScope onEnter(
-      @Advice.This final Object servlet,
-      @Advice.Argument(0) final ServletRequest request,
-      @Advice.Argument(1) final ServletResponse response) {
+  public static SpanScopePair onEnter(
+      @Advice.This final Object servlet, @Advice.Argument(0) final ServletRequest request) {
+    final Span current = TRACER.getCurrentSpan();
+    final boolean hasActiveTrace = current != null && current.getContext().isValid();
```

Comment:
```suggestion
    final boolean hasActiveTrace = current.getContext().isValid();
```
TRACER.getCurrentSpan() never returns `null`